### PR TITLE
Assignment: Writing Tests

### PR DESCRIPTION
**Pull request by alexhcole95**
## Summary

Completed writing tests assignment with working calculator and applicable testing file.

## Changes

1. Completed assignment.
2. Created calculator directory with calculator.go and calculator_test.go files.
3. Created main.go file for calculator usage.

### CODE REVIEW

File: calculator/calculator_test.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
+package calculator_test
+
+import (
+	"github.com/alexhcole95/pivottechschool/calculator"
+	"testing"
+)
+
+func TestAdd(t *testing.T) {
+	result := calculator.Add(1, 2)
```
Comment (jboursiquot) [inline calculator/calculator_test.go:9]: Use the table-driven testing technique.

File: calculator/calculator_test.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+
+func TestMultiply(t *testing.T) {
+	result := calculator.Multiply(5, 3)
+	if result != 15 {
+		t.Error("Expected 15, got ", result)
+	}
+}
+
+func TestDivide(t *testing.T) {
+	result, err := calculator.Divide(6, 3)
```
Comment (jboursiquot) [inline calculator/calculator_test.go:30]: You need to also test the divide-by-zero case. Do so within a table-driven setup.

File: calculator/calculator_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+func TestSubtract(t *testing.T) {
+	result := calculator.Subtract(5, 3)
+	if result != 2 {
+		t.Error("Expected 2, got ", result)
+	}
+}
+
+func TestMultiply(t *testing.T) {
+	result := calculator.Multiply(5, 3)
```
Comment (jboursiquot) [inline calculator/calculator_test.go:23]: Make table-driven.

File: main.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+	"bufio"
+	"errors"
+	"fmt"
+	"github.com/alexhcole95/pivottechschool/calculator"
+	"os"
+	"strconv"
+	"strings"
+)
+
+func main() {
```
Comment (jboursiquot) [inline main.go:13]: You're doing a lot more than asked for by the assignment.

It's okay because you're learning and you're doing so in class but as a professional, you want to solve the problem you have, no more, no less. Any additional effort needs to be a proposal to your team that has buy-in before you proceed with the actual work. This is because code maintenance has a cost. Any unnecessary code you add to a project or codebase adds to that cost.

**Review by jboursiquot — CHANGES_REQUESTED** [review]
Good start. In addition to the inline comments, do the following:

- Put your `main.go` in a `cmd/calculator` folder. Follow the convention for executables in Go.
- You should rename your readme.md to README.md. Again, a convention that reduces the cognitive load on everybody.

File: calculator/calculator_test.go (hunk 1 above)
Comment (alexhcole95) [reply calculator/calculator_test.go]: Hey Johnny, 

Aside from renaming the readme.md to README.md, removing the HTML tags, and moving the main.go file I believe I did a table-driven testing technique. I did some Googling and found some stuff on my own (along with your video) and think I may have got it. Let me know if that looks right though. I appreciate your feedback and patience!

**Review by jboursiquot — APPROVED** [review]
Nice work.
